modules/renter/hostdb/hostweight.go

```diff
@@ -31,113 +33,115 @@ var (
 		}
-		if age < 2000 {
-			weight = weight.Div64(4) // 16x total
+		if entry.RemainingStorage < 50*requiredStorage {
+			weight = weight.Div64(4) // 24x total penalty
 		}
-		if age < 1000 {
-			weight = weight.Div64(4) // 64x total
+		if entry.RemainingStorage < 25*requiredStorage {
+			weight = weight.Div64(5) // 95x total penalty
 		}
-		if age < 288 {
-			weight = weight.Div64(10) // 640x total
+		if entry.RemainingStorage < 10*requiredStorage {
+			weight = weight.Div64(6) // 570x total penalty
+		}
+		if entry.RemainingStorage < 5*requiredStorage {
+			weight = weight.Div64(10) // 5,700x total penalty
+		}
+		if entry.RemainingStorage < requiredStorage {
+			weight = weight.Div64(100) // 570,000 total penalty
 		}
-	} else {
-		// Shouldn't happen, but the usecase is covered anyway.
-		weight = weight.Div64(1000) // Because something weird is happening, don't trust this host very much.
-	}
 
-	// Account for collateral. Collateral has a somewhat complicated
-	// relationship with price, because raising the collateral inherently
-	// raises the price for renters. If the host's score increases linearly to
-	// the amount of collateral they put up, then the host will gain score from
-	// increasing collateral until the siafund fee makes up about 15% of the
-	// total price. After that, the host will actually lose points for having
-	// too much collateral.
-	//
-	// The renter has control over how much collateral the host uses.
-	// Currently, this control is not implemented, so the hosts are penalized
-	// for setting very high collateral values. Once the renter is clamping the
-	// amount being spent on collateral, the hostdb can also clamp the amount
-	// of collateral being taken into account by the host, to optimize the
-	// host's score for the renter's needs.
-	if entry.Collateral.IsZero() {
-		// Instead of zeroing out the weight, just return the weight as though
-		// the collateral is 1 hasting. Competitively speaking, this is
-		// effectively zero.
+		// Enact penalities for hosts running older versions.
+		if build.VersionCmp(entry.Version, "1.0.3") < 0 {
+			weight = weight.Div64(5) // 5x total penalty.
+		}
+		if build.VersionCmp(entry.Version, "1.0.0") < 0 {
+			weight = weight.Div64(10) // 50x total penalty.
+		}
+
+		// Enact penalities for newer hosts, as it's less certain that they will
+		// have reliable uptime.
+		if hdb.blockHeight >= entry.FirstSeen {
+			age := hdb.blockHeight - entry.FirstSeen
+			if age < 6000 {
+				weight = weight.Div64(2) // 2x total
+			}
+			if age < 4000 {
+				weight = weight.Div64(2) // 4x total
+			}
+			if age < 2000 {
+				weight = weight.Div64(4) // 16x total
+			}
+			if age < 1000 {
+				weight = weight.Div64(4) // 64x total
+			}
+			if age < 288 {
+				weight = weight.Div64(10) // 640x total
+			}
+		} else {
+			// Shouldn't happen, but the usecase is covered anyway.
+			weight = weight.Div64(1000) // Because something weird is happening, don't trust this host very much.
```

Comment:
Should probably be 10e3, not 1e3. Should definitely use e notation since it's got >= three zeroes